### ISSUE TITLE
massage various expected errors for test deployments

### ISSUE THREAD

**Issue by glycerine** [cockroachdb/cockroach#3060]
Making a separate issue from the busy-loop issue of #3054, as these errors/warnings in the log feel like distinct issues.

Summary: There seem to be three possibly serious errors/warnings when I try to start a one-node cluster. Abbreviated notes here, full details below:

1. `the raft group doesn't exist for range 1`

2. `http: TLS handshake error from 10.0.0.165:54925: EOF`

3. unable to allocate a target store;  `E1109 09:49:11.613995 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available`

OSX 10.10.5 on go1.5.1 and cockroachdb at tip / fb4fff8fb8177c070e927993d8fff62b63f608fb

The 'fresh' script shows how to reproduce this.

~~~
jaten@Jasons-MacBook-Pro:~/go/src/github.com/cockroachdb/cockroach$ cat ./fresh
cat ./fresh
#!/bin/bash

rm -rf certs
./cockroach cert create-ca
./cockroach cert create-node 10.0.0.165
./cockroach cert create-client root
rm -rf /Users/jaten/data
./cockroach init --stores=hdd=/Users/jaten/data
./cockroach start --stores=hdd=/Users/jaten/data --gossip=10.0.0.165:26257 --addr=10.0.0.165:26257

jaten@Jasons-MacBook-Pro:~/go/src/github.com/cockroachdb/cockroach$ ./fresh
./fresh
I1109 09:48:51.450364 87479 server/context.go:175  initialized 1 storage engine(s)
I1109 09:48:51.450631 87479 storage/engine/rocksdb.go:103  opening rocksdb instance at "/Users/jaten/data"
I1109 09:48:51.459953 87479 multiraft/multiraft.go:936  node 1 campaigning because initial confstate is [1]
I1109 09:48:51.460688 87479 storage/replica_command.go:1141  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1447091332.460s
I1109 09:48:51.461464 87479 cli/start.go:84  cockroach cluster 42dd893c-c831-4a71-9814-3f209ad76a7a has been initialized
I1109 09:48:51.461544 87479 storage/engine/rocksdb.go:134  closing rocksdb instance at "/Users/jaten/data"
I1109 09:48:51.495151 87480 cli/start.go:114  build Vers: go1.5.1
I1109 09:48:51.495311 87480 cli/start.go:115  build Tag:  alpha-5821-gfb4fff8
I1109 09:48:51.495326 87480 cli/start.go:116  build Time: 2015/11/09 17:47:52
I1109 09:48:51.495339 87480 cli/start.go:117  build Deps: /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/biogo/store:3b4c041f52c224ee4a44f5c8b150d003a40643a0 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cockroachdb/c-lz4:c40aaae2fc50293eb8750b34632bc3efe813e23f /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cockroachdb/c-protobuf:6a18bfcdd5169966cd1235edb71d749c950216e8 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cockroachdb/c-rocksdb:bf15ead80bdc205a19b3d33415b23c156a3cf371 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cockroachdb/c-snappy:5c6d0932e0adaffce4bfca7bdf2ac37f79952ccf /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cockroachdb/cockroach:fb4fff8fb8177c070e927993d8fff62b63f608fb /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/coreos/etcd:2fe6893d5d71e2d91df27f72814aeccc747f4f64 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/cpuguy83/go-md2man:71acacd42f85e5e82f70a55327789582a5200a90 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/elazarl/go-bindata-assetfs:d5cac425555ca5cf00694df246e04f05e6a55150 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/gogo/protobuf:04145330265d088c218da8916c50eab3814c76a0 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/google/btree:cc6329d4279e3f025a53a83c397d2339b5705c45 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/inconshreveable/mousetrap:76626ae9c91c4f2a10f34cad8ce83ea42c93bb75 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/julienschmidt/httprouter:77a895ad01ebc98a4dc95d8355bc825ce80a56f6 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/montanaflynn/stats:94c87312234200f782148dd8de4855d53a8c9c5d /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/olekukonko/tablewriter:a5eefc286b03d5560735698ef36c83728a6ae560 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/peterh/liner:b850cf8c6d0ee52309aad09ac610508c6c75e819 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/russross/blackfriday:d4ee3ea08bf2f1471dbf7846090d9dd931609968 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/shurcooL/sanitized_anchor_name:10ef21a441db47d8b13ebcc5fd2310f636973c77 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/spf13/cobra:2ab15e2b40dbbe9e48d556cb90fb95b8f1f3f105 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/github.com/spf13/pflag:08b1a584251b5b62f458943640fc8ebd4d50aaa5 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/golang.org/x/crypto:c8b9e6388ef638d5a8a9d865c634befdc46a6784 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/golang.org/x/net:2fd7f1556cc0a4aa45f8aee9626ce303433f31eb /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/golang.org/x/text:0fe7e6856182a6ebfcf1e6a7aa90bead9a8e1bc0 /Users/jaten/dev/cnet/go/src/causalnetworks.com/src/gopkg.in/yaml.v1:9f9df34309c04878acc86042b16630b0f696e1de
I1109 09:48:51.601619 87480 server/context.go:175  initialized 1 storage engine(s)
I1109 09:48:51.601702 87480 cli/start.go:146  starting cockroach cluster
I1109 09:48:51.605336 87480 storage/engine/rocksdb.go:103  opening rocksdb instance at "/Users/jaten/data"
I1109 09:48:51.610964 87480 server/node.go:286  initialized store store=1:1 ([hdd]=/Users/jaten/data): {Capacity:499055067136 Available:356340715520 RangeCount:0}
I1109 09:48:51.658772 87480 server/node.go:213  node ID 1 initialized
I1109 09:48:51.659003 87480 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"tcp" address_field:"10.0.0.165:26257" > attrs:<> 
I1109 09:48:51.659057 87480 server/node.go:376  connecting to gossip network to verify cluster ID...
I1109 09:48:51.659090 87480 server/node.go:393  node connected via gossip and verified as part of cluster "42dd893c-c831-4a71-9814-3f209ad76a7a"
I1109 09:48:51.659175 87480 server/node.go:251  Started node with [[hdd]=/Users/jaten/data] engine(s) and attributes []
I1109 09:48:51.659218 87480 server/server.go:205  starting https server at 10.0.0.165:26257
W1109 09:48:56.611661 87480 storage/raft_log_queue.go:111  storage/raft_log_queue.go:74: the raft group doesn't exist for range 1
I1109 09:48:56.612109 87480 multiraft/multiraft.go:936  node 1 campaigning because initial confstate is [1]
E1109 09:48:56.613194 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available
I1109 09:48:56.630986 87480 server.go:1934  http: TLS handshake error from 10.0.0.165:54925: EOF
E1109 09:49:01.613449 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available
I1109 09:49:01.662088 87480 server/status/runtime.go:122  runtime stats: 48 goroutines, 1.19MiB active, 0.00cgo/sec, 0.00/0.00 %(u/s)time, 0.00 %gc (1x)
E1109 09:49:06.613772 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available
I1109 09:49:09.326609 87480 server.go:1934  http: TLS handshake error from 10.0.0.165:55109: EOF
E1109 09:49:11.613995 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available
I1109 09:49:11.662850 87480 server/status/runtime.go:122  runtime stats: 47 goroutines, 2.76MiB active, 50.50cgo/sec, 0.00/0.00 %(u/s)time, 0.00 %gc (0x)
E1109 09:49:16.615084 87480 storage/queue.go:368  failure processing replica range=1 [""-"\xff\xff") from replicate queue: storage/allocator.go:186: unable to allocate a target store; no candidates available
  C-c C-cI1109 09:49:18.723789 87480 cli/start.go:171  initiating graceful shutdown of server
I1109 09:49:18.724004 87480 storage/engine/rocksdb.go:134  closing rocksdb instance at "/Users/jaten/data"
I1109 09:49:18.724268 87480 cli/start.go:195  server drained and shutdown completed
jaten@Jasons-MacBook-Pro:~/go/src/github.com/cockroachdb/cockroach$ 
~~~

**Comment by tamird** [comment]
As explained in #3054, `3. unable to allocate a target store;` is not a bug.

@BramGruneir can you look into `1.`?

I have no idea what `2.` is about, though.

**Comment by glycerine** [comment]
> As explained in #3054, 3. unable to allocate a target store; is not a bug.

Running a single node cluster should be completely viable without encountering errors, no? Since the program knows there is only a single node, it should not complain when there is only a single node. 

Hence I do think it is a (low priority) bug.

**Comment by tamird** [comment]
You can silence the error by updating the zone config to replicate the data to just a single node. That said, better error plumbing is possible here, and was discussed in #1565:

> I'm talking about returning an error from `replicateQueue.process`, not `allocator.AllocateTarget`. When `AllocateTarget` cannot find a target, `process` should not return an error.

> Even if `replicateQueue` doesn't do anything useful with error messages returned by `AllocateTarget`, they can still be useful: we will have some diagnostic UI where an admin can inspect an underreplicated range and we want to be able to report more details like "there are not enough nodes" or "nodes exist but none match requested constraints". It might make sense to separate that into a separate UI-only method, though, instead of constructing a message just to throw it away.

**Comment by tschottdorf** [comment]
> I1109 09:49:09.326609 87480 server.go:1934  http: TLS handshake error from 10.0.0.165:55109: EOF

any chance this is you trying to access the node via `http` instead of `https`?

**Comment by glycerine** [comment]
>> I1109 09:49:09.326609 87480 server.go:1934 http: TLS handshake error from 10.0.0.165:55109: EOF
>any chance this is you trying to access the node via http instead of https?

I'm not sure what it means when chrome shows the "https" with the strikethrough line in it, but there were, at the very least, some kind of certificate problem -- certainly. I'll attach a screenshot of what happens when I try to connect from chrome to https://<current_localhost_ip:cockroach_db_port>. I *think* this means it is doing https (not http) but that the certificate was signed by a CA that my browser doesn't trust.

<img width="880" alt="https_fails_cockroachdb" src="https://cloud.githubusercontent.com/assets/445247/11065073/2b70bb78-8776-11e5-9677-0203c855fa6a.png">

In case the screenshot gets lots, chrome (Chrome Version 46.0.2490.80 (64-bit) on osx 10.10.5) complains "NET::ERR_CERT_AUTHORITY_INVALID" at which point I have to click through the (unsafe) link to get to the cockroachdb admin page.

**Comment by mberhault** [comment]
This is the standard chrome warning when encountering self-signed certificates. Chrome has no way to verify the CA or server certs, so it's taking the (imho reasonable) action of prompting you for verification.

Proceeding to "unsafe" will take you to the page.
Alternatively, you can manually add the generated CA cert to chrome, but that's rather tedious. If chrome attempted to hit plain http on the backend, that would explain the EOF warning you were seeing earlier. Unfortunately, SSL handshake issues are often very obscure, with little information in the errors themselves.

**Comment by glycerine** [comment]
Here's the log
~~~
I1110 06:37:13.852801 93643 server.go:1934  http: TLS handshake error from 10.239.11.43:59875: EOF
I1110 06:37:13.859451 93643 server.go:1934  http: TLS handshake error from 10.239.11.43:59876: EOF
I1110 06:37:14.065818 93643 server.go:1934  http: TLS handshake error from 10.239.11.43:59877: EOF
~~~

and then on a later page load from a new private browser instance gives
~~~
I1110 06:59:17.021512 93643 server.go:1934  http: TLS handshake error from 10.239.11.43:59992: EOF
~~~

The only thing I see in the chrome console failure to load the favicon.ico
"Failed to load resource: the server responded with a status of 500 (Internal Server Error)   ... https://10.239.11.43:26257/favicon.ico "

I'm not sure if that could be the cause?

**Comment by tschottdorf** [comment]
We should make sure that expected/informational messages in the early live stages of a cluster don't look like "hard" errors.

@BramGruneir, care to take care of the `the raft group doesn't exist for range 1`?

**Comment by BramGruneir** [comment]
@tschottdorf Sure, can do.

**Comment by knz** [comment]
@glycerine I have picked up this issue and checked the status of the messages you reported initially with this issue. Here is the status:

1. the first error about the "raft group" has been addressed by #3590.
2. the second error about EOF in http connections has been greatly alleviated already, I can't seem to reproduce. If you could check anew that would help.
3. about the missing replica message, as you've seen in #3054 and others we find it quite important to highlight the need to run CockroachDB with multiple nodes. This is documented in the README since https://github.com/lpambagoda/cockroach/commit/f8bd059b5febaae2387576a3d238ff4de214a079; I also marked your issue #4192 so that its solution makes it to our documentation.

So unless you still see something unwieldy about the http/EOF error, I suggest we can close this particular issue. What do you think?

**Comment by glycerine** [comment]
I won't be able to try this again until the end the week. It certainly seems reasonable to me to close it out based on the significant work done. Anything residual would surely be a new issue.
